ddl/rollingback.go

```diff
@@ -157,11 +157,33 @@ func rollingbackAddColumns(t *meta.Meta, job *model.Job) (ver int64, err error)
 }
 
 func rollingbackDropColumn(t *meta.Meta, job *model.Job) (ver int64, err error) {
-	tblInfo, colInfo, err := checkDropColumn(t, job)
+	tblInfo, colInfo, idxInfos, err := checkDropColumn(t, job)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
 
+	if len(idxInfos) > 0 {
+		for _, indexInfo := range idxInfos {
+			originalState := indexInfo.State
+			switch indexInfo.State {
+			case model.StateWriteOnly, model.StateDeleteOnly, model.StateDeleteReorganization, model.StateNone:
+				// We can not rollback now, so just continue to drop index.
+				// Normally won't fetch here, because there is check when cancel ddl jobs. see function: isJobRollbackable.
```

Comment:
In `IsJobRollbackable`, when the state is None, it's rollbackable.

Reply:
This code is reference `rollingbackDropIndex`, in this function StateNone is not rollbackable.

Reply:
I will update the comments.